go/l1client/ethereum_client.go

```diff
@@ -0,0 +1,209 @@
+package l1client
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/obscuronet/obscuro-playground/go/l1client/rollupcontractlib"
+	"github.com/obscuronet/obscuro-playground/go/l1client/wallet"
+	"github.com/obscuronet/obscuro-playground/go/log"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+var connectionTimeout = 15 * time.Second
+
+type ethNode struct {
+	client          *ethclient.Client           // the underlying eth rpc client
+	id              common.Address              // TODO remove the id common.Address
+	wallet          wallet.Wallet               // wallet containing thea ccount information // TODO this does not need to be coupled together
```

Comment:
@otherview Typo - should be "containing the account information"

Reply:
Done 👍